src/exec.cpp

```diff
@@ -746,6 +746,13 @@ static proc_performer_t get_performer_for_process(process_t *p, const job_t *job
         };
     } else {
         assert(p->type == process_type_t::function);
+        // If this function is the only process in the current job
```

Comment:
This is the only real change- everything else is a consequence of moving job_t::job_id to a private member variable and introducing a public getter.